### PR TITLE
Support for new autogenerated OBS Dependencies (WIP)

### PR DESCRIPTION
**Pull request by Xaymar**
### Description
Modifies the necessary files to support the new automatically generated obs dependencies archives. These are slightly different from what used to be in dependencies.zip due to changes in binary or linked names.

A test win-deps.7z is available at: https://github.com/Xaymar/obs-deps/suites/283034446/artifacts/186217. Note that at the time of writing this, some dependencies are still missing: python3, swig, websockets. These dependencies need to be copied from the original dependencies.zip for the time until they are added.

### Motivation and Context
The `dependencies.zip` content randomly changes and relies on someone not messing up. It is far better to have all the dependencies automatically generated from CI, so I went ahead and made [a PR to obs-deps](https://github.com/obsproject/obs-deps/pull/2), which does exactly that.

### How Has This Been Tested?
- Tested streaming SRT with PR #1748 applied, OBS-OBS, OBS-ffplay and ffmpeg-ffplay streaming worked without any major flaws. No crashes or new bugs here.
- Added the obs-ffmpeg-encoder plugin, which I used for H264 and H265 encoding tests (relies on ffmpeg).

### Types of changes
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Performance enhancement (non-breaking change which improves efficiency)
- Breaking change (fix or feature that would cause existing functionality to change)

### Dependencies:
- [ ] [Automatically generated obs-deps for Windows](https://github.com/obsproject/obs-deps/pull/2)

### Checklist:
- [ ] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [ ] All commit messages are properly formatted and commits squashed where appropriate.

### CODE REVIEW

File: plugins/obs-outputs/librtmp/rtmp.c [hunk 1]
```diff
@@ -55,6 +55,7 @@
 #include <mbedtls/ctr_drbg.h>
 #include <mbedtls/md5.h>
 #include <mbedtls/base64.h>
+#include <mbedtls/x509_crt.h>
```
Comment (kkartaltepe) [inline plugins/obs-outputs/librtmp/rtmp.c]: This was added in support for newer mbedtls, but does it exist in older mbedtls? Does it need to be wrapped in an ifdef?

Comment (Xaymar) [reply plugins/obs-outputs/librtmp/rtmp.c:58]: It exists in 2.11 (current dependencies.zip) too, I'm not actually sure if it's needed. My MSVC just kept complaining about something_x509_crt_t not being defined, so I added it.

**Comment by Xaymar** [comment]
This is currently stalled over the holidays as I want to enjoy my time with friends and family. Will pick it back up after Christmas is over.

**Comment by Xaymar** [comment]
Closing as I have no further interest in this matter.

**Pull request closed without merging**